File: VB/WebSite/App_Code/CustomAppointmentFormController.vb
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.Web.ASPxScheduler.Internal
Imports DevExpress.XtraScheduler


Public Class CustomAppointmentFormController
    Inherits AppointmentFormController

    Public Sub New(ByVal control As ASPxScheduler, ByVal apt As Appointment)
        MyBase.New(control, apt)
    End Sub

    Public Property CustomInfoField() As String
        Get
            Return CStr(EditedAppointmentCopy.CustomFields("ApptCustomInfo"))
        End Get
        Set(ByVal value As String)
            EditedAppointmentCopy.CustomFields("ApptCustomInfo") = value
        End Set
    End Property

    Private Property SourceCustomInfoField() As String
        Get
            Return CStr(SourceAppointment.CustomFields("ApptCustomInfo"))
        End Get
        Set(ByVal value As String)
            SourceAppointment.CustomFields("ApptCustomInfo") = value
        End Set
    End Property

    Public Overrides Function IsAppointmentChanged() As Boolean
        Dim isChanged As Boolean = MyBase.IsAppointmentChanged()
        Return isChanged OrElse SourceCustomInfoField <> CustomInfoField
    End Function
End Class

File: CS/WebSite/App_Code/CustomAppointmentFormController.cs
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.ASPxScheduler.Internal;
using DevExpress.XtraScheduler;


public class CustomAppointmentFormController : AppointmentFormController {
    public CustomAppointmentFormController(ASPxScheduler control, Appointment apt)
        : base(control, apt) { }

    public string CustomInfoField {
        get { return (string)EditedAppointmentCopy.CustomFields["ApptCustomInfo"]; }
        set { EditedAppointmentCopy.CustomFields["ApptCustomInfo"] = value; }
    }

    string SourceCustomInfoField {
        get { return (string)SourceAppointment.CustomFields["ApptCustomInfo"]; }
        set { SourceAppointment.CustomFields["ApptCustomInfo"] = value; }
    }

    public override bool IsAppointmentChanged() {
        bool isChanged = base.IsAppointmentChanged();
        return isChanged || SourceCustomInfoField != CustomInfoField;
    }
}
